### ISSUE TITLE
CI Tests failing

### ISSUE THREAD

**Issue by gr2m** [hoodiehq/hoodie-app-tracker#32]
Our tests currently tend to fail on travis with something like this:

<img width="521" alt="screen shot 2016-03-05 at 17 20 02" src="https://cloud.githubusercontent.com/assets/39992/13550909/93ad3a28-e2f6-11e5-93d9-b0111c24dbae.png">

The test that caused this error is [.check() when request errors](https://github.com/hoodiehq/hoodie-app-tracker/blob/76d55a10881106bd45ad7e92fb0de552dcd6c5bf/test/connection-status-test.js#L73-L92). The assertion `expect(events.length).to.equal(1)` is the one failing, events.length is `0` sometimes. 

Restarting the build on Travis usually helps, but this random error smells like a race condition issue, and we should resolve it. 

To run the tests locally the same way they are run on Travis, run this command in your the folder of the `hoodie-app-tracker` repository

```bash
CLIENT=saucelabs:chrome SAUCELABS_USERNAME=hoodie SAUCELABS_ACCESS_KEY=1f6164de-f3d1-4af9-83d0-5358b42fbe56 ./node_modules/.bin/frontend-test-background mocha test/connection-status-test.js
```

I’ve tried it several times but they always passed :/ 

That’s how far I got with my investigations, maybe someone else can find out more?

**Comment by finnp** [comment]
On my computer it also failed the first time. Most of the time it works though. But sometimes I get:

```
  1) hoodie.connectionStatus .check() when request errors:

      undefined: expected 0 to equal 1
      + expected - actual

      -0
      +1



  2) hoodie.connectionStatus .check() when request succeeds after it failed before:

      undefined: expected 0 to equal 1
      + expected - actual

      -0
      +1
```

**Comment by gr2m** [comment]
if can find out why that happens sometimes, or how to reliably reproduce the issue, that’d be amazing :)

**Comment by gr2m** [comment]
most likely resolved by #44, but let’s leave it open for a few weeks to see if the failed tests happen again. Close this issue if it’s April ’16 or later :)
